include/drivers/flash.h

```diff
@@ -161,12 +161,31 @@ static inline int z_impl_flash_erase(struct device *dev, off_t offset,
  *  @brief  Enable or disable write protection for a flash memory
  *
  *  This API is required to be called before the invocation of write or erase
- *  API. Please note that on some flash components, the write protection is
+ *  API. Behaviour at calls to @ref flash_write @ref flash_erase without
+ *  disabling the flash protection prior is undefined.
```

Comment:
s/prior/first/
s/undefined/driver-dependent/

"undefined behavior" is a very bad thing in C; it allows for "I'm gonna publish your passwords on the Internet".   "Implementation-defined behavior" is something we could work with.  In this case the implementation is the driver.  The documented return value for the functions is the generic "negative errno code on fail", so "driver-dependent" allows write/erase without unprotect to either fail (as had been required before) or to succeed (as with the code change part of this).

Reply:
I was intended to emphasize that flash_protection must be disable before each write/erase operation. This is not something like in C - mean that some driver might work without this call prior, but not using flash_protection is like waiting for luck. maybe worth to redact this differently.

Reply:
@nvlsianpu can you address the grammar issues that @pabigot raised?

Reply:
done